solr/core/src/java/org/apache/solr/handler/admin/api/SnapshotAPI.java

```diff
@@ -0,0 +1,218 @@
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import org.apache.lucene.index.IndexCommit;
+import org.apache.solr.api.JerseyResource;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.params.CoreAdminParams;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.core.IndexDeletionPolicyWrapper;
+import org.apache.solr.core.SolrCore;
+import org.apache.solr.core.snapshots.SolrSnapshotManager;
+import org.apache.solr.core.snapshots.SolrSnapshotMetaDataManager;
+import org.apache.solr.jersey.PermissionName;
+import org.apache.solr.jersey.SolrJerseyResponse;
+
+@Path("/cores/{coreName}/snapshots")
+public class SnapshotAPI extends JerseyResource {
+
+  private final CoreContainer coreContainer;
+
+  @Inject
+  public SnapshotAPI(CoreContainer coreContainer) {
+    this.coreContainer = coreContainer;
+  }
+
+  @POST
+  @Path("/{snapshotName}")
+  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @PermissionName(CORE_EDIT_PERM)
+  public CreateSnapshotResponse createSnapshot(
+      @Parameter(description = "The name of the core to snapshot.", required = true)
+          @PathParam("coreName")
+          String coreName,
+      @Parameter(description = "The name to associate with the core snapshot.", required = true)
+          @PathParam("snapshotName")
+          String snapshotName)
+      throws Exception {
+    final CreateSnapshotResponse response = instantiateJerseyResponse(CreateSnapshotResponse.class);
+
+    try (SolrCore core = coreContainer.getCore(coreName)) {
+      if (core == null) {
+        throw new SolrException(
+            SolrException.ErrorCode.BAD_REQUEST, "Unable to locate core " + coreName);
+      }
+
+      final String indexDirPath = core.getIndexDir();
+      final IndexDeletionPolicyWrapper delPol = core.getDeletionPolicy();
+      final IndexCommit ic = delPol.getAndSaveLatestCommit();
+      try {
+        if (null == ic) {
+          throw new SolrException(
+              SolrException.ErrorCode.BAD_REQUEST,
+              "No index commits to snapshot in core " + coreName);
+        }
+        final SolrSnapshotMetaDataManager mgr = core.getSnapshotMetaDataManager();
+        mgr.snapshot(snapshotName, indexDirPath, ic.getGeneration());
+
+        response.core = core.getName();
+        response.commitName = snapshotName;
+        response.indexDirPath = indexDirPath;
+        response.generation = ic.getGeneration();
+        response.files = ic.getFileNames();
+      } finally {
+        delPol.releaseCommitPoint(ic);
+      }
+    }
+
+    return response;
+  }
+
+  public static class CreateSnapshotResponse extends SolrJerseyResponse {
+    @JsonProperty(CoreAdminParams.CORE)
```

Comment:
[0] It's not required, especially in cases where the nuances of what a particular field represents are unclear, but if you DO have a good idea for any of these fields and are willing, you can use the `@Schema` annotation to document it.

These descriptions end up in the OpenAPI spec generated by the Solr build, which we'll use in subsequent releases to create Solr clients for different languages (e.g. golang, Python).  We're also hoping to use them to auto-generate certain sections of our ref-guide.  So lots of indirect value there if you're able to document any of these.

Reply:
I'll make an effort to add some Schema statements. It seems like a good idea to add those now while we're working on this API.